pyirf/simulations.py

```diff
@@ -168,6 +169,115 @@ def calculate_n_showers_per_energy_and_fov(self, energy_bins, fov_bins):
+            The expected number of events inside each of the
+            ``energy_bins``, ``fov_offset_bins`` and ``fov_position_angle_bins``.
+            Dimension (n_energy_bins, n_fov_offset_bins, n_fov_position_angle_bins)
+            This is a floating point number.
+            The actual numbers will follow a poissionian distribution around this
+            expected value.
+        """
+        e_fov_offset_integral = self.calculate_n_showers_per_energy_and_fov(
+            energy_bins, fov_offset_bins
+        )
+        position_angle_integral = np.full(
+            len(fov_position_angle_bins) - 1,
+            self.calculate_n_showers_per_fov(
+                np.linspace(self.viewcone_min, self.viewcone_max, 2)
+            )
+            / (len(fov_position_angle_bins) - 1),
+        )
+
+        return (
+            e_fov_offset_integral[:, :, np.newaxis]
+            * position_angle_integral
+            / self.n_showers
+        )
+
+    @u.quantity_input(
+        energy_bins=u.TeV, fov_longitude_bins=u.deg, fov_latitude_bins=u.rad
+    )
+    def calculate_n_showers_3d_lonlat(
+        self, energy_bins, fov_longitude_bins, fov_latitude_bins
+    ):
+        """
+        Calculate number of showers that were simulated in the given
+        energy and 2D fov bins in nominal coordinates.
+
+        This assumes the events were generated uniformly distributed per solid angle,
+        and from a powerlaw in energy like CORSIKA simulates events.
+
+        Parameters
+        ----------
+        energy_bins: astropy.units.Quantity[energy]
+            The energy bin edges for which to calculate the number of simulated showers
+        fov_longitude_bins: astropy.units.Quantity[angle]
+            The FOV longitude bin edges for which to calculate the number of simulated showers
+        fov_latitude_bins: astropy.units.Quantity[angle]
+            The FOV latitude bin edges for which to calculate the number of simulated showers
+
+        Returns
+        -------
+        n_showers: numpy.ndarray(ndim=3)
+            The expected number of events inside each of the
+            ``energy_bins``, ``fov_longitude_bins`` and ``fov_latitude_bins``.
+            Dimension (n_energy_bins, n_fov_longitude_bins, n_fov_latitude_bins)
+            This is a floating point number.
+            The actual numbers will follow a poissionian distribution around this
+            expected value.
+        """
+        fov_bin_midpoints_lon, fov_bin_midpoints_lat = np.meshgrid(
+            (fov_longitude_bins[:-1] + fov_longitude_bins[1:]) / 2,
+            (fov_latitude_bins[:-1] + fov_latitude_bins[1:]) / 2,
+        )
+        mask_outside_fov = np.logical_or(
+            np.sqrt(fov_bin_midpoints_lon**2 + fov_bin_midpoints_lat**2)
+            > self.viewcone_max,
+            np.sqrt(fov_bin_midpoints_lon**2 + fov_bin_midpoints_lat**2)
+            < self.viewcone_min,
+        )
+        A_bins = (
```

Comment:
Please implement a new function in `util` to compute the solid angle for these rectangular bins and use it here.

We will need it also for #276